src/app/Library/Datatable/Datatable.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\Datatable;
+
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanel;
+use Backpack\CRUD\CrudManager;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\View\Component;
+
+class Datatable extends Component
+{
+    protected string $tableId;
+
+    public function __construct(
+        private string $controller,
+        private ?CrudPanel $crud = null,
+        private bool $updatesUrl = true,
+        private ?\Closure $configure = null,
+        private ?string $type = null,
+        private ?string $name = null
+    ) {
+        // Set active controller for proper context
+        CrudManager::setActiveController($controller);
+
+        $this->crud ??= CrudManager::crudFromController($controller);
+
+        $controllerPart = str_replace('\\', '_', $this->controller);
+        $typePart = $this->type ?? 'default';
+        $namePart = $this->name ?? 'default';
+        $uniqueId = md5($controllerPart.'_'.$typePart.'_'.$namePart);
+        $this->tableId = 'crudTable_'.$uniqueId;
```

Comment:
If all these lines are just to set the tableId, why not clean this up and create a function for it?

```suggestion
        $this->crud ??= CrudManager::crudFromController($controller);
        $this->tableId = $this->generateTableId();
```